File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    static bool cmp(const pair<int, int>& a, const pair<int, int>& b) {
        if (a.first == b.first) {
            return a.second < b.second;
        }
        return a.first < b.first;
    }
    vector<int> twoSum(vector<int>& nums, int target) {

        int n = nums.size();
        vector<pair<int, int>> sorted;

        for (int i = 0; i < n; i++) {
            sorted.push_back({nums[i], i});
        }

        sort(sorted.begin(), sorted.end(), cmp);
        int left = 0, right = n - 1;

        while (left < right) {
            int sum = sorted[left].first + sorted[right].first;

            if (sum == target)
                return {sorted[left].second, sorted[right].second};
            else if (sum > target)
                right--;
            else
                left++;
        }

        return {-1, -1};
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer>map=new HashMap<>();
        int n=nums.length;
        
        for(int i=0;i<n;i++){
            if(map.containsKey(target-nums[i])){
                int index=map.get(target-nums[i]);
                return new int[]{index,i};
            }
            map.put(nums[i],i);
        }

        return new int[]{-1,-1};
    }
}